core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/aggregation/modes/ofRowExpression.kt

```diff
@@ -75,7 +75,8 @@ internal inline fun <T, reified C, reified R> Grouped<T>.aggregateOf(
     val type = typeOf<R>()
     return aggregateInternal {
         val value = aggregator.aggregateOf(df, expression)
-        yield(path, value, type, null, false)
+        val inferType = !aggregator.preservesType
```

Comment:
sorry, could not parse the logic here, could you please explain or add a comment in the code

Reply:
Rewrote it now with `calculateReturnType`. 
But the gist of this specific overload `aggregateOf` is that it executes the calls like `meanOf`, `minOf`, etc, so the result of calling a `statisticOf { rowExpression }` function from within a `GroupBy`.
The result of calling the aggregator on the dataframe with the rowExpression here is a single `value: R`, which will be "yielded" with a certain path and type. This makes it so that `GroupBy` can, say, store it in a new `mean` column :). It's very high level, but hopefully the idea is a bit clearer now.